---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Cmd+Shift+Enter*. 

## Step 1: import allele frequency data
```{r}
library("Biostrings", lib.loc="/Library/Frameworks/R.framework/Versions/3.3/Resources/library")

file_name = "Ribp_UVp" # or "Ribp_UVm"
dat <- read.csv(paste(file_name,".csv",sep=""),header=TRUE)

# check coverage
for (i in 1:10){
  coverage <- dat[,(i*6)]
  print(below_cov <- which(coverage<100))
}
# time sampled
t=7
```

## Step 2: calculate new mutations at each sampling time point
```{r}
####### P0 to P5
# check major allele
#max_first <- pmax(dat$P0_A,dat$P0_C,dat$P0_G,dat$P0_T,na.rm=TRUE)
major_A <- which(dat$P0_A > 1)
major_C <- which(dat$P0_C > 1)
major_G <- which(dat$P0_G > 1)
major_T <- which(dat$P0_T > 1)
# check minor allele
#max_last <- pmax(dat$P5_A,dat$P5_C,dat$P5_G,dat$P5_T,na.rm=TRUE)
minor_A <- which(dat$P5_A > 1)
minor_C <- which(dat$P5_C > 1)
minor_G <- which(dat$P5_G > 1)
minor_T <- which(dat$P5_T > 1)

aboveerr_A <- minor_A[which(!(minor_A %in%  major_A))] # index of new mutations -> alleles present in minor_A not in major_A
dat[aboveerr_A,1] # position of new mutations

aboveerr_C <- minor_C[which(!(minor_C %in%  major_C))]
dat[aboveerr_C,1]

aboveerr_G <- minor_G[which(!(minor_G %in%  major_G))]
dat[aboveerr_G,1]

aboveerr_T <- minor_T[which(!(minor_T %in%  major_T))]
dat[aboveerr_T,1]

# check major allele
max_first <- pmax(dat$P0_A,dat$P0_C,dat$P0_G,dat$P0_T,na.rm=TRUE)
major_A <- which(dat$P0_A == max_first)
major_C <- which(dat$P0_C == max_first)
major_G <- which(dat$P0_G == max_first)
major_T <- which(dat$P0_T == max_first)
# create data with only major and minor alleles
new_dat <- data.frame()
colnames(dat)[41] # 7*6-1
seq=1:7
# Major
Ma_A <- as.matrix(cbind(rep("A",length(major_A)),dat[major_A,seq*6-4]))
Ma_C <- as.matrix(cbind(rep("C",length(major_C)),dat[major_C,seq*6-3]))
Ma_G <- as.matrix(cbind(rep("G",length(major_G)),dat[major_G,seq*6-2]))
Ma_T <- as.matrix(cbind(rep("T",length(major_T)),dat[major_T,seq*6-1]))
colnames(Ma_A) <- NULL
colnames(Ma_C) <- NULL
colnames(Ma_G) <- NULL
colnames(Ma_T) <- NULL
new_dat <- rbind(Ma_A,Ma_C,Ma_G,Ma_T)
new_dat_Ma <- new_dat[order(as.numeric(rownames(new_dat))),]

### sequences at t for A
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for A
UV_dat_Ma[aboveerr_A] <- "A"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_A_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_A_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="0_5_to_A_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="0_5_to_A_2.txt", Matrix=NA, block.width=50)

### sequences at t for C
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for c
UV_dat_Ma[aboveerr_C] <- "C"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_C_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_C_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="0_5_to_C_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="0_5_to_C_2.txt", Matrix=NA, block.width=50)

### sequences at t for G
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for G
UV_dat_Ma[aboveerr_G] <- "G"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_G_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_G_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="0_5_to_G_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="0_5_to_G_2.txt", Matrix=NA, block.width=50)

### sequences at t for T
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for T
UV_dat_Ma[aboveerr_T] <- "T"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_T_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_T_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="0_5_to_T_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="0_5_to_T_2.txt", Matrix=NA, block.width=50)

## Synonymous mutations
(length(aboveerr_A)+length(aboveerr_C)+length(aboveerr_G)+length(aboveerr_T)) - (NS_A_1+NS_A_2+NS_C_1+NS_C_2+NS_G_1+NS_G_2+NS_T_1+NS_T_2)
```

```{r}
####### P5 to P7
# check major allele
#max_first <- pmax(dat$P5_A,dat$P5_C,dat$P5_G,dat$P5_T,na.rm=TRUE)
major_A <- which(dat$P5_A > 1)
major_C <- which(dat$P5_C > 1)
major_G <- which(dat$P5_G > 1)
major_T <- which(dat$P5_T > 1)
# check minor allele
#max_last <- pmax(dat$P7_A,dat$P7_C,dat$P7_G,dat$P7_T,na.rm=TRUE)
minor_A <- which(dat$P7_A > 1)
minor_C <- which(dat$P7_C > 1)
minor_G <- which(dat$P7_G > 1)
minor_T <- which(dat$P7_T > 1)

aboveerr_A <- minor_A[which(!(minor_A %in%  major_A))] # index of new mutations -> alleles present in minor_A not in major_A
dat[aboveerr_A,1] # position of new mutations

aboveerr_C <- minor_C[which(!(minor_C %in%  major_C))]
dat[aboveerr_C,1]

aboveerr_G <- minor_G[which(!(minor_G %in%  major_G))]
dat[aboveerr_G,1]

aboveerr_T <- minor_T[which(!(minor_T %in%  major_T))]
dat[aboveerr_T,1]

# check major allele
max_first <- pmax(dat$P5_A,dat$P5_C,dat$P5_G,dat$P5_T,na.rm=TRUE)
major_A <- which(dat$P5_A == max_first)
major_C <- which(dat$P5_C == max_first)
major_G <- which(dat$P5_G == max_first)
major_T <- which(dat$P5_T == max_first)
# create data with only major and minor alleles
new_dat <- data.frame()
colnames(dat)[41] # 7*6-1
seq=1:7
# Major
Ma_A <- as.matrix(cbind(rep("A",length(major_A)),dat[major_A,seq*6-4]))
Ma_C <- as.matrix(cbind(rep("C",length(major_C)),dat[major_C,seq*6-3]))
Ma_G <- as.matrix(cbind(rep("G",length(major_G)),dat[major_G,seq*6-2]))
Ma_T <- as.matrix(cbind(rep("T",length(major_T)),dat[major_T,seq*6-1]))
colnames(Ma_A) <- NULL
colnames(Ma_C) <- NULL
colnames(Ma_G) <- NULL
colnames(Ma_T) <- NULL
new_dat <- rbind(Ma_A,Ma_C,Ma_G,Ma_T)
new_dat_Ma <- new_dat[order(as.numeric(rownames(new_dat))),]

### sequences at t for A
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for A
UV_dat_Ma[aboveerr_A] <- "A"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_A_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_A_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="5_7_to_A_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="5_7_to_A_2.txt", Matrix=NA, block.width=50)

### sequences at t for C
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for c
UV_dat_Ma[aboveerr_C] <- "C"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_C_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_C_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="5_7_to_C_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="5_7_to_C_2.txt", Matrix=NA, block.width=50)

### sequences at t for G
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for G
UV_dat_Ma[aboveerr_G] <- "G"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_G_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_G_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="5_7_to_G_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="5_7_to_G_2.txt", Matrix=NA, block.width=50)

### sequences at t for T
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for T
UV_dat_Ma[aboveerr_T] <- "T"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_T_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_T_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="5_7_to_T_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="5_7_to_T_2.txt", Matrix=NA, block.width=50)

## Synonymous mutations
(length(aboveerr_A)+length(aboveerr_C)+length(aboveerr_G)+length(aboveerr_T)) - (NS_A_1+NS_A_2+NS_C_1+NS_C_2+NS_G_1+NS_G_2+NS_T_1+NS_T_2)
```

```{r}
####### P7 to P10
# check major allele
#max_first <- pmax(dat$P7_A,dat$P7_C,dat$P7_G,dat$P7_T,na.rm=TRUE)
major_A <- which(dat$P7_A > 1)
major_C <- which(dat$P7_C > 1)
major_G <- which(dat$P7_G > 1)
major_T <- which(dat$P7_T > 1)
# check minor allele
#max_last <- pmax(dat$P10_A,dat$P10_C,dat$P10_G,dat$P10_T,na.rm=TRUE)
minor_A <- which(dat$P10_A > 1)
minor_C <- which(dat$P10_C > 1)
minor_G <- which(dat$P10_G > 1)
minor_T <- which(dat$P10_T > 1)

aboveerr_A <- minor_A[which(!(minor_A %in%  major_A))] # index of new mutations -> alleles present in minor_A not in major_A
dat[aboveerr_A,1] # position of new mutations

aboveerr_C <- minor_C[which(!(minor_C %in%  major_C))]
dat[aboveerr_C,1]

aboveerr_G <- minor_G[which(!(minor_G %in%  major_G))]
dat[aboveerr_G,1]

aboveerr_T <- minor_T[which(!(minor_T %in%  major_T))]
dat[aboveerr_T,1]

# check major allele
max_first <- pmax(dat$P7_A,dat$P7_C,dat$P7_G,dat$P7_T,na.rm=TRUE)
major_A <- which(dat$P7_A == max_first)
major_C <- which(dat$P7_C == max_first)
major_G <- which(dat$P7_G == max_first)
major_T <- which(dat$P7_T == max_first)
# create data with only major and minor alleles
new_dat <- data.frame()
colnames(dat)[41] # 7*6-1
seq=1:7
# Major
Ma_A <- as.matrix(cbind(rep("A",length(major_A)),dat[major_A,seq*6-4]))
Ma_C <- as.matrix(cbind(rep("C",length(major_C)),dat[major_C,seq*6-3]))
Ma_G <- as.matrix(cbind(rep("G",length(major_G)),dat[major_G,seq*6-2]))
Ma_T <- as.matrix(cbind(rep("T",length(major_T)),dat[major_T,seq*6-1]))
colnames(Ma_A) <- NULL
colnames(Ma_C) <- NULL
colnames(Ma_G) <- NULL
colnames(Ma_T) <- NULL
new_dat <- rbind(Ma_A,Ma_C,Ma_G,Ma_T)
new_dat_Ma <- new_dat[order(as.numeric(rownames(new_dat))),]

### sequences at t for A
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for A
UV_dat_Ma[aboveerr_A] <- "A"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_A_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_A_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="7_10_to_A_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="7_10_to_A_2.txt", Matrix=NA, block.width=50)

### sequences at t for C
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for c
UV_dat_Ma[aboveerr_C] <- "C"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_C_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_C_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="7_10_to_C_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="7_10_to_C_2.txt", Matrix=NA, block.width=50)

### sequences at t for G
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for G
UV_dat_Ma[aboveerr_G] <- "G"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_G_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_G_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="7_10_to_G_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="7_10_to_G_2.txt", Matrix=NA, block.width=50)

### sequences at t for T
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for T
UV_dat_Ma[aboveerr_T] <- "T"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_T_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_T_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="7_10_to_T_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="7_10_to_T_2.txt", Matrix=NA, block.width=50)

## Synonymous mutations
(length(aboveerr_A)+length(aboveerr_C)+length(aboveerr_G)+length(aboveerr_T)) - (NS_A_1+NS_A_2+NS_C_1+NS_C_2+NS_G_1+NS_G_2+NS_T_1+NS_T_2)
```

```{r}
####### P10 to P15
# check major allele
#max_first <- pmax(dat$P10_A,dat$P10_C,dat$P10_G,dat$P10_T,na.rm=TRUE)
major_A <- which(dat$P10_A > 1)
major_C <- which(dat$P10_C > 1)
major_G <- which(dat$P10_G > 1)
major_T <- which(dat$P10_T > 1)
# check minor allele
#max_last <- pmax(dat$P15_A,dat$P15_C,dat$P15_G,dat$P15_T,na.rm=TRUE)
minor_A <- which(dat$P15_A > 1)
minor_C <- which(dat$P15_C > 1)
minor_G <- which(dat$P15_G > 1)
minor_T <- which(dat$P15_T > 1)

aboveerr_A <- minor_A[which(!(minor_A %in%  major_A))] # index of new mutations -> alleles present in minor_A not in major_A
dat[aboveerr_A,1] # position of new mutations

aboveerr_C <- minor_C[which(!(minor_C %in%  major_C))]
dat[aboveerr_C,1]

aboveerr_G <- minor_G[which(!(minor_G %in%  major_G))]
dat[aboveerr_G,1]

aboveerr_T <- minor_T[which(!(minor_T %in%  major_T))]
dat[aboveerr_T,1]

# check major allele
max_first <- pmax(dat$P10_A,dat$P10_C,dat$P10_G,dat$P10_T,na.rm=TRUE)
major_A <- which(dat$P10_A == max_first)
major_C <- which(dat$P10_C == max_first)
major_G <- which(dat$P10_G == max_first)
major_T <- which(dat$P10_T == max_first)
# create data with only major and minor alleles
new_dat <- data.frame()
colnames(dat)[41] # 7*6-1
seq=1:7
# Major
Ma_A <- as.matrix(cbind(rep("A",length(major_A)),dat[major_A,seq*6-4]))
Ma_C <- as.matrix(cbind(rep("C",length(major_C)),dat[major_C,seq*6-3]))
Ma_G <- as.matrix(cbind(rep("G",length(major_G)),dat[major_G,seq*6-2]))
Ma_T <- as.matrix(cbind(rep("T",length(major_T)),dat[major_T,seq*6-1]))
colnames(Ma_A) <- NULL
colnames(Ma_C) <- NULL
colnames(Ma_G) <- NULL
colnames(Ma_T) <- NULL
new_dat <- rbind(Ma_A,Ma_C,Ma_G,Ma_T)
new_dat_Ma <- new_dat[order(as.numeric(rownames(new_dat))),]

### sequences at t for A
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for A
UV_dat_Ma[aboveerr_A] <- "A"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_A_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_A_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="10_15_to_A_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="10_15_to_A_2.txt", Matrix=NA, block.width=50)

### sequences at t for C
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for c
UV_dat_Ma[aboveerr_C] <- "C"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_C_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_C_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="10_15_to_C_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="10_15_to_C_2.txt", Matrix=NA, block.width=50)

### sequences at t for G
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for G
UV_dat_Ma[aboveerr_G] <- "G"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_G_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_G_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="10_15_to_G_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="10_15_to_G_2.txt", Matrix=NA, block.width=50)

### sequences at t for T
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for T
UV_dat_Ma[aboveerr_T] <- "T"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_T_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_T_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="10_15_to_T_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="10_15_to_T_2.txt", Matrix=NA, block.width=50)

## Synonymous mutations
(length(aboveerr_A)+length(aboveerr_C)+length(aboveerr_G)+length(aboveerr_T)) - (NS_A_1+NS_A_2+NS_C_1+NS_C_2+NS_G_1+NS_G_2+NS_T_1+NS_T_2)
```

```{r}
####### P15 to P18
# check major allele
#max_first <- pmax(dat$P15_A,dat$P15_C,dat$P15_G,dat$P15_T,na.rm=TRUE)
major_A <- which(dat$P15_A > 1)
major_C <- which(dat$P15_C > 1)
major_G <- which(dat$P15_G > 1)
major_T <- which(dat$P15_T > 1)
# check minor allele
#max_last <- pmax(dat$P18_A,dat$P18_C,dat$P18_G,dat$P18_T,na.rm=TRUE)
minor_A <- which(dat$P18_A > 1)
minor_C <- which(dat$P18_C > 1)
minor_G <- which(dat$P18_G > 1)
minor_T <- which(dat$P18_T > 1)

aboveerr_A <- minor_A[which(!(minor_A %in%  major_A))] # index of new mutations -> alleles present in minor_A not in major_A
dat[aboveerr_A,1] # position of new mutations

aboveerr_C <- minor_C[which(!(minor_C %in%  major_C))]
dat[aboveerr_C,1]

aboveerr_G <- minor_G[which(!(minor_G %in%  major_G))]
dat[aboveerr_G,1]

aboveerr_T <- minor_T[which(!(minor_T %in%  major_T))]
dat[aboveerr_T,1]

# check major allele
max_first <- pmax(dat$P15_A,dat$P15_C,dat$P15_G,dat$P15_T,na.rm=TRUE)
major_A <- which(dat$P15_A == max_first)
major_C <- which(dat$P15_C == max_first)
major_G <- which(dat$P15_G == max_first)
major_T <- which(dat$P15_T == max_first)
# create data with only major and minor alleles
new_dat <- data.frame()
colnames(dat)[41] # 7*6-1
seq=1:7
# Major
Ma_A <- as.matrix(cbind(rep("A",length(major_A)),dat[major_A,seq*6-4]))
Ma_C <- as.matrix(cbind(rep("C",length(major_C)),dat[major_C,seq*6-3]))
Ma_G <- as.matrix(cbind(rep("G",length(major_G)),dat[major_G,seq*6-2]))
Ma_T <- as.matrix(cbind(rep("T",length(major_T)),dat[major_T,seq*6-1]))
colnames(Ma_A) <- NULL
colnames(Ma_C) <- NULL
colnames(Ma_G) <- NULL
colnames(Ma_T) <- NULL
new_dat <- rbind(Ma_A,Ma_C,Ma_G,Ma_T)
new_dat_Ma <- new_dat[order(as.numeric(rownames(new_dat))),]

### sequences at t for A
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for A
UV_dat_Ma[aboveerr_A] <- "A"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_A_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_A_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="15_18_to_A_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="15_18_to_A_2.txt", Matrix=NA, block.width=50)

### sequences at t for C
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for c
UV_dat_Ma[aboveerr_C] <- "C"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_C_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_C_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="15_18_to_C_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="15_18_to_C_2.txt", Matrix=NA, block.width=50)

### sequences at t for G
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for G
UV_dat_Ma[aboveerr_G] <- "G"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_G_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_G_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="15_18_to_G_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="15_18_to_G_2.txt", Matrix=NA, block.width=50)

### sequences at t for T
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for T
UV_dat_Ma[aboveerr_T] <- "T"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_T_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_T_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="15_18_to_T_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="15_18_to_T_2.txt", Matrix=NA, block.width=50)

## Synonymous mutations
(length(aboveerr_A)+length(aboveerr_C)+length(aboveerr_G)+length(aboveerr_T)) - (NS_A_1+NS_A_2+NS_C_1+NS_C_2+NS_G_1+NS_G_2+NS_T_1+NS_T_2)
```

```{r}
####### P18 to P20
# check major allele
#max_first <- pmax(dat$P18_A,dat$P18_C,dat$P18_G,dat$P18_T,na.rm=TRUE)
major_A <- which(dat$P18_A > 1)
major_C <- which(dat$P18_C > 1)
major_G <- which(dat$P18_G > 1)
major_T <- which(dat$P18_T > 1)
# check minor allele
#max_last <- pmax(dat$P20_A,dat$P20_C,dat$P20_G,dat$P20_T,na.rm=TRUE)
minor_A <- which(dat$P20_A > 1)
minor_C <- which(dat$P20_C > 1)
minor_G <- which(dat$P20_G > 1)
minor_T <- which(dat$P20_T > 1)

aboveerr_A <- minor_A[which(!(minor_A %in%  major_A))] # index of new mutations -> alleles present in minor_A not in major_A
dat[aboveerr_A,1] # position of new mutations

aboveerr_C <- minor_C[which(!(minor_C %in%  major_C))]
dat[aboveerr_C,1]

aboveerr_G <- minor_G[which(!(minor_G %in%  major_G))]
dat[aboveerr_G,1]

aboveerr_T <- minor_T[which(!(minor_T %in%  major_T))]
dat[aboveerr_T,1]

# check major allele
max_first <- pmax(dat$P18_A,dat$P18_C,dat$P18_G,dat$P18_T,na.rm=TRUE)
major_A <- which(dat$P18_A == max_first)
major_C <- which(dat$P18_C == max_first)
major_G <- which(dat$P18_G == max_first)
major_T <- which(dat$P18_T == max_first)
# create data with only major and minor alleles
new_dat <- data.frame()
colnames(dat)[41] # 7*6-1
seq=1:7
# Major
Ma_A <- as.matrix(cbind(rep("A",length(major_A)),dat[major_A,seq*6-4]))
Ma_C <- as.matrix(cbind(rep("C",length(major_C)),dat[major_C,seq*6-3]))
Ma_G <- as.matrix(cbind(rep("G",length(major_G)),dat[major_G,seq*6-2]))
Ma_T <- as.matrix(cbind(rep("T",length(major_T)),dat[major_T,seq*6-1]))
colnames(Ma_A) <- NULL
colnames(Ma_C) <- NULL
colnames(Ma_G) <- NULL
colnames(Ma_T) <- NULL
new_dat <- rbind(Ma_A,Ma_C,Ma_G,Ma_T)
new_dat_Ma <- new_dat[order(as.numeric(rownames(new_dat))),]

### sequences at t for A
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for A
UV_dat_Ma[aboveerr_A] <- "A"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_A_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_A_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_A_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_A_2.txt", Matrix=NA, block.width=50)

### sequences at t for C
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for c
UV_dat_Ma[aboveerr_C] <- "C"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_C_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_C_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_C_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_C_2.txt", Matrix=NA, block.width=50)

### sequences at t for G
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for G
UV_dat_Ma[aboveerr_G] <- "G"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_G_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_G_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_G_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_G_2.txt", Matrix=NA, block.width=50)

### sequences at t for T
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for T
UV_dat_Ma[aboveerr_T] <- "T"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_T_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_T_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_T_1.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_T_2.txt", Matrix=NA, block.width=50)

## Synonymous mutations
(length(aboveerr_A)+length(aboveerr_C)+length(aboveerr_G)+length(aboveerr_T)) - (NS_A_1+NS_A_2+NS_C_1+NS_C_2+NS_G_1+NS_G_2+NS_T_1+NS_T_2)
```

```{r}
####### P18 to R2_P20
# check major allele
#max_first <- pmax(dat$P18_A,dat$P18_C,dat$P18_G,dat$P18_T,na.rm=TRUE)
major_A <- which(dat$P18_A > 1)
major_C <- which(dat$P18_C > 1)
major_G <- which(dat$P18_G > 1)
major_T <- which(dat$P18_T > 1)
# check minor allele
#max_last <- pmax(dat$R2_P20_A,dat$R2_P20_C,dat$R2_P20_G,dat$R2_P20_T,na.rm=TRUE)
minor_A <- which(dat$R2_P20_A > 1)
minor_C <- which(dat$R2_P20_C > 1)
minor_G <- which(dat$R2_P20_G > 1)
minor_T <- which(dat$R2_P20_T > 1)

aboveerr_A <- minor_A[which(!(minor_A %in%  major_A))] # index of new mutations -> alleles present in minor_A not in major_A
dat[aboveerr_A,1] # position of new mutations

aboveerr_C <- minor_C[which(!(minor_C %in%  major_C))]
dat[aboveerr_C,1]

aboveerr_G <- minor_G[which(!(minor_G %in%  major_G))]
dat[aboveerr_G,1]

aboveerr_T <- minor_T[which(!(minor_T %in%  major_T))]
dat[aboveerr_T,1]

# check major allele
max_first <- pmax(dat$P18_A,dat$P18_C,dat$P18_G,dat$P18_T,na.rm=TRUE)
major_A <- which(dat$P18_A == max_first)
major_C <- which(dat$P18_C == max_first)
major_G <- which(dat$P18_G == max_first)
major_T <- which(dat$P18_T == max_first)
# create data with only major and minor alleles
new_dat <- data.frame()
colnames(dat)[41] # 7*6-1
seq=1:7
# Major
Ma_A <- as.matrix(cbind(rep("A",length(major_A)),dat[major_A,seq*6-4]))
Ma_C <- as.matrix(cbind(rep("C",length(major_C)),dat[major_C,seq*6-3]))
Ma_G <- as.matrix(cbind(rep("G",length(major_G)),dat[major_G,seq*6-2]))
Ma_T <- as.matrix(cbind(rep("T",length(major_T)),dat[major_T,seq*6-1]))
colnames(Ma_A) <- NULL
colnames(Ma_C) <- NULL
colnames(Ma_G) <- NULL
colnames(Ma_T) <- NULL
new_dat <- rbind(Ma_A,Ma_C,Ma_G,Ma_T)
new_dat_Ma <- new_dat[order(as.numeric(rownames(new_dat))),]

### sequences at t for A
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for A
UV_dat_Ma[aboveerr_A] <- "A"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_A_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_A_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_A_1_R2.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_A_2_R2.txt", Matrix=NA, block.width=50)

### sequences at t for C
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for c
UV_dat_Ma[aboveerr_C] <- "C"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_C_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_C_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_C_1_R2.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_C_2_R2.txt", Matrix=NA, block.width=50)

### sequences at t for G
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for G
UV_dat_Ma[aboveerr_G] <- "G"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_G_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_G_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_G_1_R2.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_G_2_R2.txt", Matrix=NA, block.width=50)

### sequences at t for T
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for T
UV_dat_Ma[aboveerr_T] <- "T"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_T_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_T_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_T_1_R2.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_T_2_R2.txt", Matrix=NA, block.width=50)

## Synonymous mutations
(length(aboveerr_A)+length(aboveerr_C)+length(aboveerr_G)+length(aboveerr_T)) - (NS_A_1+NS_A_2+NS_C_1+NS_C_2+NS_G_1+NS_G_2+NS_T_1+NS_T_2)
```

```{r}
####### P18 to R3_P20
# check major allele
#max_first <- pmax(dat$P18_A,dat$P18_C,dat$P18_G,dat$P18_T,na.rm=TRUE)
major_A <- which(dat$P18_A > 1)
major_C <- which(dat$P18_C > 1)
major_G <- which(dat$P18_G > 1)
major_T <- which(dat$P18_T > 1)
# check minor allele
#max_last <- pmax(dat$R3_P20_A,dat$R3_P20_C,dat$R3_P20_G,dat$R3_P20_T,na.rm=TRUE)
minor_A <- which(dat$R3_P20_A > 1)
minor_C <- which(dat$R3_P20_C > 1)
minor_G <- which(dat$R3_P20_G > 1)
minor_T <- which(dat$R3_P20_T > 1)

aboveerr_A <- minor_A[which(!(minor_A %in%  major_A))] # index of new mutations -> alleles present in minor_A not in major_A
dat[aboveerr_A,1] # position of new mutations

aboveerr_C <- minor_C[which(!(minor_C %in%  major_C))]
dat[aboveerr_C,1]

aboveerr_G <- minor_G[which(!(minor_G %in%  major_G))]
dat[aboveerr_G,1]

aboveerr_T <- minor_T[which(!(minor_T %in%  major_T))]
dat[aboveerr_T,1]

# check major allele
max_first <- pmax(dat$P18_A,dat$P18_C,dat$P18_G,dat$P18_T,na.rm=TRUE)
major_A <- which(dat$P18_A == max_first)
major_C <- which(dat$P18_C == max_first)
major_G <- which(dat$P18_G == max_first)
major_T <- which(dat$P18_T == max_first)
# create data with only major and minor alleles
new_dat <- data.frame()
colnames(dat)[41] # 7*6-1
seq=1:7
# Major
Ma_A <- as.matrix(cbind(rep("A",length(major_A)),dat[major_A,seq*6-4]))
Ma_C <- as.matrix(cbind(rep("C",length(major_C)),dat[major_C,seq*6-3]))
Ma_G <- as.matrix(cbind(rep("G",length(major_G)),dat[major_G,seq*6-2]))
Ma_T <- as.matrix(cbind(rep("T",length(major_T)),dat[major_T,seq*6-1]))
colnames(Ma_A) <- NULL
colnames(Ma_C) <- NULL
colnames(Ma_G) <- NULL
colnames(Ma_T) <- NULL
new_dat <- rbind(Ma_A,Ma_C,Ma_G,Ma_T)
new_dat_Ma <- new_dat[order(as.numeric(rownames(new_dat))),]

### sequences at t for A
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for A
UV_dat_Ma[aboveerr_A] <- "A"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_A_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_A_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_A_1_R3.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_A_2_R3.txt", Matrix=NA, block.width=50)

### sequences at t for C
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for c
UV_dat_Ma[aboveerr_C] <- "C"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_C_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_C_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_C_1_R3.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_C_2_R3.txt", Matrix=NA, block.width=50)

### sequences at t for G
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for G
UV_dat_Ma[aboveerr_G] <- "G"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_G_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_G_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_G_1_R3.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_G_2_R3.txt", Matrix=NA, block.width=50)

### sequences at t for T
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for T
UV_dat_Ma[aboveerr_T] <- "T"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_T_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_T_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_T_1_R3.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_T_2_R3.txt", Matrix=NA, block.width=50)

## Synonymous mutations
(length(aboveerr_A)+length(aboveerr_C)+length(aboveerr_G)+length(aboveerr_T)) - (NS_A_1+NS_A_2+NS_C_1+NS_C_2+NS_G_1+NS_G_2+NS_T_1+NS_T_2)
```

```{r}
####### P18 to R4_P20
# check major allele
#max_first <- pmax(dat$P18_A,dat$P18_C,dat$P18_G,dat$P18_T,na.rm=TRUE)
major_A <- which(dat$P18_A > 1)
major_C <- which(dat$P18_C > 1)
major_G <- which(dat$P18_G > 1)
major_T <- which(dat$P18_T > 1)
# check minor allele
#max_last <- pmax(dat$R4_P20_A,dat$R4_P20_C,dat$R4_P20_G,dat$R4_P20_T,na.rm=TRUE)
minor_A <- which(dat$R4_P20_A > 1)
minor_C <- which(dat$R4_P20_C > 1)
minor_G <- which(dat$R4_P20_G > 1)
minor_T <- which(dat$R4_P20_T > 1)

aboveerr_A <- minor_A[which(!(minor_A %in%  major_A))] # index of new mutations -> alleles present in minor_A not in major_A
dat[aboveerr_A,1] # position of new mutations

aboveerr_C <- minor_C[which(!(minor_C %in%  major_C))]
dat[aboveerr_C,1]

aboveerr_G <- minor_G[which(!(minor_G %in%  major_G))]
dat[aboveerr_G,1]

aboveerr_T <- minor_T[which(!(minor_T %in%  major_T))]
dat[aboveerr_T,1]

# check major allele
max_first <- pmax(dat$P18_A,dat$P18_C,dat$P18_G,dat$P18_T,na.rm=TRUE)
major_A <- which(dat$P18_A == max_first)
major_C <- which(dat$P18_C == max_first)
major_G <- which(dat$P18_G == max_first)
major_T <- which(dat$P18_T == max_first)
# create data with only major and minor alleles
new_dat <- data.frame()
colnames(dat)[41] # 7*6-1
seq=1:7
# Major
Ma_A <- as.matrix(cbind(rep("A",length(major_A)),dat[major_A,seq*6-4]))
Ma_C <- as.matrix(cbind(rep("C",length(major_C)),dat[major_C,seq*6-3]))
Ma_G <- as.matrix(cbind(rep("G",length(major_G)),dat[major_G,seq*6-2]))
Ma_T <- as.matrix(cbind(rep("T",length(major_T)),dat[major_T,seq*6-1]))
colnames(Ma_A) <- NULL
colnames(Ma_C) <- NULL
colnames(Ma_G) <- NULL
colnames(Ma_T) <- NULL
new_dat <- rbind(Ma_A,Ma_C,Ma_G,Ma_T)
new_dat_Ma <- new_dat[order(as.numeric(rownames(new_dat))),]

### sequences at t for A
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for A
UV_dat_Ma[aboveerr_A] <- "A"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_A_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_A_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_A_1_R4.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_A_2_R4.txt", Matrix=NA, block.width=50)

### sequences at t for C
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for c
UV_dat_Ma[aboveerr_C] <- "C"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_C_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_C_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_C_1_R4.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_C_2_R4.txt", Matrix=NA, block.width=50)

### sequences at t for G
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for G
UV_dat_Ma[aboveerr_G] <- "G"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_G_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_G_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_G_1_R4.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_G_2_R4.txt", Matrix=NA, block.width=50)

### sequences at t for T
UV_dat_Ma <- new_dat_Ma[,1]
UV_Ma_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Ma_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
### sequence at t+1 for T
UV_dat_Ma[aboveerr_T] <- "T"
UV_Mi_1_1 <- DNAString(paste(UV_dat_Ma[(752-45):(4867-45)],collapse=""))
UV_Mi_1_2 <- DNAString(paste(UV_dat_Ma[(4886-61):(7336-61)],collapse=""))
## Amino acid global alignment
NS_T_1 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"))
NS_T_2 <- nmismatch(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"))
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_1), translate(UV_Mi_1_1), substitutionMatrix = "BLOSUM50"), file="18_20_to_T_1_R4.txt", Matrix=NA, block.width=50)
writePairwiseAlignments(pairwiseAlignment(translate(UV_Ma_1_2), translate(UV_Mi_1_2), substitutionMatrix = "BLOSUM50"), file="18_20_to_T_2_R4.txt", Matrix=NA, block.width=50)

## Synonymous mutations
(length(aboveerr_A)+length(aboveerr_C)+length(aboveerr_G)+length(aboveerr_T)) - (NS_A_1+NS_A_2+NS_C_1+NS_C_2+NS_G_1+NS_G_2+NS_T_1+NS_T_2)
```

